### ISSUE TITLE
Icon: fix padding between icons

### ISSUE THREAD

**Issue by gretanausedaite** [iTwin/iTwinUI#1520]
Imagine `*` is an icon and `.` is a space.
We want a list of icons to be something like this:

`. . * . . * . . * . .` 

To achieve this we need to update icon css code.

**Comment by mayank99** [comment]
some code that i was playing around:
```scss
padding-inline: var(--_iui-svg-padding-inline);

+ & {
  padding-inline-start: 0;
}
```

**Comment by FlyersPh9** [comment]
I'm thinking the values should match [the borderless button's `--_iui-button-padding-inline`](https://github.com/iTwin/iTwinUI/blob/ddafcdde472a9a4009864c0471a310fb0b3219c7/packages/itwinui-css/src/button/base.scss#L94) for each size.

Small:
```scss
--_iui-button-padding-inline: var(--iui-size-2xs);
```

Medium:
```scss
--_iui-button-padding-inline: var(--iui-size-xs);
```

Large:
```scss
--_iui-button-padding-inline: var(--iui-size-s);
```

**Comment by mayank99** [comment]
@FlyersPh9  That was the initial thought but it looks inconsistent when multiple paddings get combined.

Since this issue is mainly for [InputWithDecorations](https://itwin.github.io/iTwinUI/dev/react/?path=/story/input-inputwithdecorations--basic)/[SearchBox](https://itwin.github.io/iTwinUI/dev/react/?path=/story/input-searchbox--basic), can you verify the values and show (visually) the spacing for:
- the leftmost/rightmost icons and the border
- the gap between two adjacent icons
- the padding between leftmost/rightmost icon and input text

I think it's ok if we need to diverge from the borderless button padding to make some of these gaps look nicer.

**Comment by FlyersPh9** [comment]
> That was the initial thought but it looks inconsistent when multiple paddings get combined.

[Your suggestion above](https://github.com/iTwin/iTwinUI/issues/1520#issuecomment-1695870778) wouldn't solve this?  Something like:

```scss
&[data-iui-size='small'] {
  --_iui-svg-padding-inline: var(--iui-size-2xs);
}

&[data-iui-size='medium'] {
  --_iui-svg-padding-inline: var(--iui-size-xs);
}

&[data-iui-size='large'] {
  --_iui-svg-padding-inline: var(--iui-size-s);
}

+ & {
  padding-inline-start: 0;
}
```

**Comment by FlyersPh9** [comment]
> It would help with collapsing the gaps between adjacent icons, but I got the impression from your comment that you wanted to keep it consistent with buttons (where adjacent paddings add up instead of collapsing).

Oh, that is a good point.  If we knew that icons would only be grouped with other icons & buttons would only be grouped with other buttons, I would say collapse the margins on adjacent icons.  But because we don't know how people will use this, might be best to hade margins add up.

> Could you also verify the other points from my comment?

In the screenshot I set these values:
- `.iui-svg-icon[data-iui-icon-size='m'] { margin-inline: calc(var(--iui-size-xs) + 2px); }` The `+ 2px` is to account for the left & right transparent border on the borderless button.
- `input { padding-inline: var(--iui-size-2xs; }`

![Screenshot 2023-09-28 at 4 56 23 PM](https://github.com/iTwin/iTwinUI/assets/849817/a40563d9-09b9-4acb-9874-a8fe84a0fce9)

> I think it's ok if we need to diverge from the borderless button padding to make some of these gaps look nicer.

Is the main concern the red space at the start not being equal to the green in between?
![Screenshot 2023-09-28 at 5 08 10 PM](https://github.com/iTwin/iTwinUI/assets/849817/53a0e786-b60a-4df8-85c2-2212c56c2fb8)

I don't know why, but the plus buttons on the right look perfectly fine to me, but the airplane icons on the left do look a little cramped.  Maybe an optical illusion?  Maybe I coded the demo wrong?
